packages/wallet/src/Transaction/ensureValidityInterval.ts

```diff
@@ -1,8 +1,20 @@
-import { Cardano } from '@cardano-sdk/core';
+import { Cardano, Seconds } from '@cardano-sdk/core';
+
+const twoHours = Seconds(2 * 3600);
+
+export const calcTimeOffsetSlotNumber = (
+  startSlot: Cardano.Slot,
+  seconds: Seconds,
+  slotLength: Seconds
+): Cardano.Slot => Cardano.Slot(startSlot.valueOf() + seconds.valueOf() * slotLength.valueOf());
 
 export const ensureValidityInterval = (
   currentSlot: Cardano.Slot,
+  { slotLength }: Cardano.CompactGenesis,
   validityInterval?: Cardano.ValidityInterval
 ): Cardano.ValidityInterval =>
-  // Todo: Based this on slot duration, to equal 2hrs
-  ({ invalidHereafter: Cardano.Slot(currentSlot.valueOf() + 3600), ...validityInterval });
+  // 2hrs based on slotLength
```

Comment:
```suggestion
  // Default is 2hrs
```

Reply:
Maybe add an couple of test cases to cover `ensureValidityInterval`, asserting the default and ability to override.

Reply:
done: [fixup! feat(wallet): convert static TTL to be based on slotLength](https://github.com/input-output-hk/cardano-js-sdk/pull/576/commits/0e41ddffc1829232e46a91fc0ef2a8a74a3c7a6f)